client/core/core.go

```diff
@@ -6750,6 +6882,47 @@ func (c *Core) handleReconnect(host string) {
 		return
 	}
 
+	type market struct { // for book re-subscribe
+		name  string
+		base  uint32
+		quote uint32
+	}
+	dc.cfgMtx.RLock()
+	mkts := make(map[string]*market, len(dc.cfg.Markets))
+	for _, m := range dc.cfg.Markets {
+		mkts[m.Name] = &market{
+			name:  m.Name,
+			base:  m.Base,
+			quote: m.Quote,
+		}
+	}
+	dc.cfgMtx.RUnlock()
+
+	// Update the orders' selfGoverned flag according to the configured markets.
+	for _, trade := range dc.trackedTrades() {
+		// If the server's market is gone, we're on our own, otherwise we are
+		// now free to swap for this order.
+		auto := mkts[trade.mktID] == nil
+		if !auto { // market exists, now check asset config and version
+			baseCfg := dc.assetConfig(trade.Base())
+			auto = baseCfg == nil || baseCfg.Version != trade.wallets.baseWallet.version // TODO: consider allowing, handling version check in wallets
```

Comment:
I guess `xcWallet` will need a `supportedVersions []uint32`  field.

Reply:
It has `baseWallet.Info().SupportedVersions`. I need to do here what I did in `(*Core).walletSet` to check `versCompat`.